packages/teuchos/core/src/Teuchos_RCPNode.hpp

```diff
@@ -586,7 +586,7 @@ class RCPNodeTmpl : public RCPNode {
   ~RCPNodeTmpl()
     {
 #ifdef TEUCHOS_DEBUG
-      TEUCHOS_TEST_FOR_EXCEPTION( ptr_!=0, std::logic_error,
+      TEUCHOS_TEST_FOR_TERMINATION( ptr_!=0,
```

Comment:
I think I can write a unit test that triggers this termination.  That would at least provide some test of this new macro.  We want to make sure that we test this somehow.

Reply:
Thats a good idea, thanks. Just let me know when you have it.